OpenRA.Mods.Common/Traits/Render/WithCargoPipsDecoration.cs

```diff
@@ -68,8 +68,8 @@ string GetPipSequence(int i)
 				if (n < pi.Weight)
 				{
 					var sequence = Info.FullSequence;
-					if (pi.CustomPipType != null && !Info.CustomPipSequences.TryGetValue(pi.CustomPipType, out sequence))
-						Log.Write("debug", "Actor type {0} defines a custom pip type {1} that is not defined for actor type {2}".F(c.Info.Name, pi.CustomPipType, self.Info.Name));
+					if (pi.CustomPipType != null)
+						Info.CustomPipSequences.TryGetValue(pi.CustomPipType, out sequence);
```

Comment:
I haven't used TryGetValue before, so please check if it's fine to use it this way.

Reply:
Almost but not quite!

The problem is that `TryGetValue` needs to set a placeholder value (in this case null) to the out value when the key isn't found. Returning the now-null sequence will cause a crash when trying to render the pip.

The best practice is to to put the `TryGetValue` in an `if` and either directly return the value on success (if that fits the usecase) or to override the invalid out value with a suitable placeholder.

In this case, because we already have to check the CustomPipType, the first case works best:
```csharp
if (n < pi.Weight)
{
	if (pi.CustomPipType != null && Info.CustomPipSequences.TryGetValue(pi.CustomPipType, out var sequence))
		return sequence;

	return Info.FullSequence;
}
```